service/src/test/java/com/ge/predix/acs/privilege/management/dao/GraphResourceRepositoryTest.java

```diff
@@ -0,0 +1,223 @@
+        assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(2L));
+        this.resourceRepository.delete(id);
+        assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(1L));
+    }
+
+    @Test
+    public void testDeleteMultiple() {
+        List<ResourceEntity> resourceEntities = new ArrayList<>();
+        assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(0L));
+        ResourceEntity resourceEntity1 = new ResourceEntity();
+        resourceEntity1.setId(persistResource1toZone1().getId());
+        resourceEntities.add(resourceEntity1);
+        assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(2L));
+        ResourceEntity resourceEntity2 = new ResourceEntity();
+        resourceEntity2.setId(persistResource2toZone1().getId());
+        resourceEntities.add(resourceEntity2);
+        assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(3L));
+        this.resourceRepository.delete(resourceEntities);
+        assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(1L));
+    }
+
+    @Test
+    public void testDeleteAll() {
+        assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(0L));
+        persistResource1toZone1();
+        assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(2L));
+        persistResource2toZone1();
+        assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(3L));
+        this.resourceRepository.deleteAll();
+        assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(1L));
+    }
+
+    @Test
+    public void testFindAll() {
+        assertEquals(IteratorUtils.count(this.graph.vertices()), 0);
+        persistResource1toZone1();
+        assertEquals(IteratorUtils.count(this.graph.vertices()), 2);
+        persistResource2toZone1();
+        assertEquals(IteratorUtils.count(this.graph.vertices()), 3);
+
+        List<ResourceEntity> resources = this.resourceRepository.findAll();
+        assertEquals(resources.size(), 2);
+    }
+
+    @Test
+    public void testFindByZone() {
+        assertEquals(IteratorUtils.count(this.graph.vertices()), 0);
+        persistResource1toZone1();
+        assertEquals(IteratorUtils.count(this.graph.vertices()), 2);
+        persistResource2toZone1();
+        assertEquals(IteratorUtils.count(this.graph.vertices()), 3);
+
+        List<ResourceEntity> resources = this.resourceRepository.findByZone(TEST_ZONE_1);
+        assertEquals(resources.size(), 2);
+    }
+
+    @Test
+    public void testGetOrCreateZoneVertex() {
+        assertEquals(IteratorUtils.count(this.graph.vertices()), 0);
+        this.resourceRepository.getOrCreateZoneVertex(TEST_ZONE_1);
+        assertEquals(IteratorUtils.count(this.graph.vertices()), 1);
+        this.resourceRepository.getOrCreateZoneVertex(TEST_ZONE_1);
+        assertEquals(IteratorUtils.count(this.graph.vertices()), 1);
+    }
+
+    @Test
+    public void testGetByZoneAndResourceIdentifier() {
```

Comment:
Yes, those are the tests we'll have to add eventually once we begin to add capabilities for hierarchical attributes.